src/logic/relation.lean

```diff
@@ -70,6 +70,28 @@ protected def map (r : α → β → Prop) (f : α → γ) (g : β → δ) : γ
 
 variables {r : α → α → Prop} {a b c d : α}
 
+/-- If a reflexive relation `r : α → α → Prop` holds over `x y : α`,
+then it holds even if `x ≠ y`. -/
+lemma reflexive_iff_ne_imp [is_refl α r] {x y : α} :
+  r x y ↔ (x ≠ y → r x y) :=
```

Comment:
One thing I'm not sure of is whether it makes more sense to state this in reverse